File: Cotecna.Inspections/Cotecna.Inspections.Abstractions/Entities/Inspection.cs
using System;

namespace Cotecna.Inspections.Abstractions.Entities
{
    public class Inspection
    {
        public int Id { get; set; }
        public DateTime InspectionDate { get; set; }
        public string Customer { get; set; }
        public string Observations { get; set; }
        public int  Status { get; set; }
        public Inspector Inspector { get; set; }
    }
}

File: Cotecna/src/app/models/inspection.ts
export class Inspection {
    public id: number;
    public inspectionDate: Date;
    public customer: string;
    public observations: string;
    public status: string;
    public inspector: Inspector;
    constructor(attributes: Inspection = null) {
        switch (attributes.status) {
            case '1': {
               attributes.status = 'New';
               break;
            }
            case '2': {
                attributes.status = 'In Progress';
                break;
             }
             case '2': {
                attributes.status = 'Done';
                break;
             }
         }
         console.log('atributes', attributes);
         Object.assign(this, attributes);
    }
}

export class Inspector {
    public id: number;
    public inspectorName: string;

     constructor(attributes: Inspector = null) {
          Object.assign(this, attributes);
     }
}
